tests/glsl-intrinsic/shader-subgroup/shader-subgroup-clustered.slang

```diff
@@ -15,33 +15,132 @@
-    float v = 1.0f;
-    outputBuffer.data[0] = testFunctionIn2ParamPlus(subgroupClusteredAdd, 1, v, 1) &&
-                           testFunctionIn2ParamPlus(subgroupClusteredMul, 1, v, 1) &&
-                           testFunctionIn2ParamPlus(subgroupClusteredMin, 1, v, 1) &&
-                           testFunctionIn2ParamPlus(subgroupClusteredMax, 1, v, 1) &&
-                           testFunctionInInt2ParamPlus(subgroupClusteredAnd, 1, v, 1) &&
-                           testFunctionInInt2ParamPlus(subgroupClusteredOr, 1, v, 1) &&
-                           testFunctionInInt2ParamPlus(subgroupClusteredXor, 1, v, 1);
+__generic<T : __BuiltinLogicalType, let N : int>
+bool testVLogical() {
+    typealias gvec = vector<T, N>;
+
+    return true
+        && subgroupClusteredAnd(gvec(T(1)), 1) == gvec(T(1))
+        && subgroupClusteredOr(gvec(T(1)), 1) == gvec(T(1))
+        && subgroupClusteredXor(gvec(T(1)), 1) == gvec(T(1))   
+        ;     
+}
+
+bool testLogical() {
+    return true
+        && test1Logical<int>()
+        && testVLogical<int, 2>()
+        && testVLogical<int, 3>()
+        && testVLogical<int, 4>()
+        //&& test1Logical<int8_t>()
+        //&& testVLogical<int8_t, 2>()
+        //&& testVLogical<int8_t, 3>()
+        //&& testVLogical<int8_t, 4>()
+        //&& test1Logical<int16_t>()
+        //&& testVLogical<int16_t, 2>()
+        //&& testVLogical<int16_t, 3>()
+        //&& testVLogical<int16_t, 4>()
+        //&& test1Logical<int64_t>()
+        //&& testVLogical<int64_t, 2>()
+        //&& testVLogical<int64_t, 3>()
+        //&& testVLogical<int64_t, 4>()
+        && test1Logical<uint>()
+        && testVLogical<uint, 2>()
+        && testVLogical<uint, 3>()
+        && testVLogical<uint, 4>()
+        && test1Logical<bool>()
+        && testVLogical<bool, 2>()
+        && testVLogical<bool, 3>()
+        && testVLogical<bool, 4>()
+        ;
+}
 
+__generic<T : __BuiltinArithmeticType>
+bool test1Arithmetic() {
+    return true
+        && subgroupClusteredAdd(T(1), 1) == T(1)
+        && subgroupClusteredMul(T(1), 1) == T(1)
+        && subgroupClusteredMin(T(1), 1) == T(1)
+        && subgroupClusteredMax(T(1), 1) == T(1)
+        ;
+}
 
+__generic<T : __BuiltinArithmeticType, let N : int>
+bool testVArithmetic() {
+    typealias gvec = vector<T, N>;
+
+    return true
+        && subgroupClusteredAdd(gvec(T(1)), 1) == gvec(T(1))
+        && subgroupClusteredMul(gvec(T(1)), 1) == gvec(T(1))
+        && subgroupClusteredMin(gvec(T(1)), 1) == gvec(T(1))
+        && subgroupClusteredMax(gvec(T(1)), 1) == gvec(T(1))
+        ;
+}
+
+bool testArithmetic() {
+    return true
+        && test1Arithmetic<float>()
+        && testVArithmetic<float, 2>()
+        && testVArithmetic<float, 3>()
+        && testVArithmetic<float, 4>()
+        //&& test1Arithmetic<double>() // hang
+        //&& testVArithmetic<double, 2>() 
+        //&& testVArithmetic<double, 3>()
+        //&& testVArithmetic<double, 4>()
```

Comment:
I think we should get "double" type working for this task.
Can you debug and figure out what is going on?

Reply:
I will look into the issue with doubles; although the problem seems fundamental rather than due to glsl.meta 

the int8/16/64/half should all be working, and should work for HLSL already, they don't work because __requireGLSLExtension does not work with GLSL.meta (no clear error or issue) -- I know this is the case because I get the following type of errors when running tests with the type: ```glslang:  (32): error :  ' const int8_t' : required extension not requested: GL_EXT_shader_subgroup_extended_types_int8```